### ISSUE TITLE
Why random initialization of image encoding still work.

### ISSUE THREAD

**Issue by ruotianluo** [karpathy/neuraltalk2#67]
I don't know if some one have done some analysis about it.
But from the code, the image encoding part is in the cnn part, which will not be trained before fine-tuning. 

Previously I thought not training encoding at first is to train a language model first. However, the result actually shows the image information is being used during the image caption generation.

Why this kind of randomly initialized encoding can still work?

**Comment by vanpersie32** [comment]
I also have the question,it is weird,If you have the answer,please inform me

**Comment by ruotianluo** [comment]
@vanpersie32 The reason why this works, is the random projection can still preserve most of the information. If you want more theories, check Extreme Learning Machine.

**Comment by vanpersie32** [comment]
thank you,the question bother me for a long time.By the way,what is your result(cider,bleu) using the checkpoint provided by andrej?

**Comment by vanpersie32** [comment]
@ruotianluo ,I combine all training set and validation set. There are about 127800 images.I set aside 5000 for validation,5000 for testing and the rest of the images for training.
The checkpoint author provided has scores:
{"CIDEr": 1.061, "Bleu_4": 0.331, "Bleu_3": 0.434, "Bleu_2": 0.571, "Bleu_1": 0.733, "ROUGE_L": 0.555, "METEOR": 0.271}
If i train from scratch,the result is 
{"CIDEr": 0.933, "Bleu_4": 0.294, "Bleu_3": 0.397, "Bleu_2": 0.537, "Bleu_1": 0.712, "ROUGE_L": 0.519, "METEOR": 0.243}
It is incredibly high,especially score on cider.I hardly believe such a simple framework can reach so high score on cider.
